tests/tx/scripts/test_script_context.py

```diff
@@ -0,0 +1,95 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import hashlib
+from unittest.mock import Mock
+
+import pytest
+
+from hathor.transaction import Transaction, TxInput, TxOutput
+from hathor.transaction.exceptions import ScriptError
+from hathor.transaction.scripts.script_context import ScriptContext
+from hathor.transaction.scripts.sighash import SighashAll, SighashBitmask
+
+
+def test_defaults() -> None:
+    settings = Mock()
+    settings.MAX_NUM_OUTPUTS = 99
```

Comment:
I think doing it like this asserts that the code is correctly retrieving information from settings, instead of having a hardcoded value that happens to be the same as the one we have in settings. I added a test for the current value in https://github.com/HathorNetwork/hathor-core/pull/912/commits/2bf75499d3e715b87646a1b5ec0f018d4655847c, though